dashboard/http_server_head.py

```diff
@@ -99,6 +114,27 @@ def get_address(self):
         assert self.http_host and self.http_port
         return self.http_host, self.http_port
 
+    async def metrics_middleware(self, app, handler):
```

Comment:
I believe we have some sort of middleware in the dashboard (that's not implemented with middleware that catches errors from handler). I think it is `_register_route`. Can you confirm if this middleware plays well with this?

Reply:
it works well.